contractcourt/channel_arbitrator.go

```diff
@@ -3202,6 +3091,192 @@ func (c *ChannelArbitrator) checkLegacyBreach() (ArbitratorState, error) {
 	return StateContractClosed, nil
 }
 
+// sweepRequest wraps the arguments used when calling `SweepInput`.
+type sweepRequest struct {
+	// input is the input to be swept.
+	input input.Input
+
+	// params holds the sweeping parameters.
+	params sweep.Params
+}
+
+// createSweepRequest creates an anchor sweeping request for a particular
+// version (local/remote/remote pending) of the commitment.
+func (c *ChannelArbitrator) createSweepRequest(
+	anchor *lnwallet.AnchorResolution, htlcs htlcSet, anchorPath string,
+	heightHint uint32) (sweepRequest, error) {
+
+	// Use the chan id as the exclusive group. This prevents any of the
+	// anchors from being batched together.
+	exclusiveGroup := c.cfg.ShortChanID.ToUint64()
+
+	// Find the deadline for this specific anchor.
+	deadline, value, err := c.findCommitmentDeadlineAndValue(
+		heightHint, htlcs,
+	)
+	if err != nil {
+		return sweepRequest{}, err
+	}
+
+	// If we cannot find a deadline, it means there's no HTLCs at stake,
```

Comment:
I think it's possible to find a deadline and still have no HTLCs at stake since the breach retribution period is a source of deadlines. So I think this may exclude some scenarios when we have no HTLCs at stake. Is that fine or will it cause issues?